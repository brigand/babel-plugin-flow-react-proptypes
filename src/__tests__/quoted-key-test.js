const babel = require('babel-core');
const content = `
var React = require('react');

type Props = {|
  unquoted: string,
	'data-cy': string,
|};

function Comp(props: Props) {
  return <div />;
}
`;

it('quoted-key', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['@babel/env', '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;

  expect(res).toMatch(/['"]unquoted['"]/);
  expect(res).toMatch(/['"]data-cy['"]/);
  expect(res).toMatchSnapshot();

});