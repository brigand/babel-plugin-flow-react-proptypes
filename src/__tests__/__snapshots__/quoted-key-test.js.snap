// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quoted-key 1`] = `
"\\"use strict\\";

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

var React = require('react');

function Comp(props) {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
}

Comp.propTypes = {
  \\"unquoted\\": _propTypes[\\"default\\"].string.isRequired,
  \\"data-cy\\": _propTypes[\\"default\\"].string.isRequired
};"
`;
